yt/yt/server/lib/chunk_pools/unittests/ordered_chunk_pool_ut.cpp

```diff
@@ -368,6 +420,161 @@ TEST_F(TOrderedChunkPoolTest, OrderedMergeSimple)
 
 ////////////////////////////////////////////////////////////////////////////////
 
+TEST_F(TOrderedChunkPoolTest, LargeChunksPreciseSlicing)
+{
+    InitTables(
+        {true, true, true} /* isTeleportable */,
+        {false, false, false} /* isVersioned */
+    );
+
+    DataSizePerJob_ = 2_KB;
+
+    InitJobConstraints();
+
+    auto chunkA1 = CreateChunk(0, 10_KB);
+    auto chunkA2 = CreateChunk(0, 22_KB);
+    auto chunkB = CreateChunk(1, 23_KB);
+    auto chunkC = CreateChunk(2, 3_KB);
+
+    CreateChunkPool();
+
+    AddChunk(chunkA1);
+    AddChunk(chunkA2);
+    AddChunk(chunkB);
+    AddChunk(chunkC);
+
+    ChunkPool_->Finish();
+
+    ExtractOutputCookiesWhilePossible();
+    auto stripeLists = GetAllStripeLists();
+
+    EXPECT_THAT(TeleportChunks_, IsEmpty());
+    EXPECT_EQ(29u, stripeLists.size());
+
+    CheckEverything(stripeLists);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+TEST_F(TOrderedChunkPoolTest, BatchRowCountBasic)
+{
+    InitTables(
+        {true, true, true} /* isTeleportable */,
+        {false, false, false} /* isVersioned */
+    );
+
+    Options_.KeepOutputOrder = true;
+    // This should have no effect!
+    Options_.MinTeleportChunkSize = 2_KB;
+    // Nor this!
+    SamplingRate_ = 0.3;
```

Comment:
Let's add a blank line after since "Nor this!" is not related to batch row count :)